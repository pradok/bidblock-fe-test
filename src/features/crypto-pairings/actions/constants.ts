export enum actionTypes {
  CRYPTO_PAIRINGS_FETCH = 'CRYPTO/PAIRINGS_FETCH',
  CRYPTO_PAIRINGS_FETCH_SUCCESS = 'CRYPTO/PAIRINGS_FETCH_SUCCESS',
  CRYPTO_PAIRINGS_FETCH_ERROR = 'CRYPTO/PAIRINGS_FETCH_ERROR',
};
