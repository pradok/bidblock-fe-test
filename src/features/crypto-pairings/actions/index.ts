export {fetchCryptoPairings, fetchSuccess, fetchError} from './fetch';
export {actionTypes} from './constants';
