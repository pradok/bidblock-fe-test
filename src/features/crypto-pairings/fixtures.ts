import {CryptoPairingsCollection} from "./models";

const pairings: CryptoPairingsCollection = [
  [
    "tBTCUSD", 6239.9, 34.92908368, 6240, 35.7623858, -30.1, -0.0048, 6239.9, 18771.42426519, 6339.9, 6125
  ],
  [
    "tLTCUSD", 75.999, 1099.6280602, 76, 486.87405918, -1.648, -0.0212, 76, 110391.68659301, 78.66, 75.049
  ]
];

export {pairings};