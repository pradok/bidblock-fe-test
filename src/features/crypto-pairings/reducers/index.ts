import cryptoPairings, {initialState} from './crypto-pairings';

export {
  initialState,
  cryptoPairings
};
