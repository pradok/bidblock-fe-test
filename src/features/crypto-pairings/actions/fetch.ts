import { actionTypes as at } from './constants';
import { CryptoPairingsCollection, CryptoPairSymbol } from '../models';

export const fetchCryptoPairings = (symbol: CryptoPairSymbol = 'ALL') => {
  return {
    symbol,
    type: at.CRYPTO_PAIRINGS_FETCH,
  };
};

export const fetchSuccess = (result: CryptoPairingsCollection) => {
  return {
    payload: result,
    type: at.CRYPTO_PAIRINGS_FETCH_SUCCESS,
  };
};

export const fetchError = (error: Error) => {
  return {
    payload: error,
    type: at.CRYPTO_PAIRINGS_FETCH_ERROR,
  };
};