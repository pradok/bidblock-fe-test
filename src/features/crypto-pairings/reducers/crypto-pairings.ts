import {cryptoPairingsToTicker} from '../../../utils/transformer';
import {actionTypes as at} from '../actions';
import {CryptoPairingsCollectionState, CryptoPairingAction} from '../models';

export const initialState: CryptoPairingsCollectionState = {
  collection: [],
  isFetched: false,
  isLoading: false,
  tickerCollection: []
};

export default (state = initialState, action: CryptoPairingAction) => {
  switch (action.type) {
    case at.CRYPTO_PAIRINGS_FETCH:
      return {...state, isLoading: true, isFetched: false, collection: initialState.collection};
    case at.CRYPTO_PAIRINGS_FETCH_SUCCESS:
      const tickerCollection = action.payload !== undefined ? cryptoPairingsToTicker(action.payload) : [];
      console.log('tickerCollection', tickerCollection);
      return {...state, isLoading: false, isFetched: true, collection: action.payload, tickerCollection};
    case at.CRYPTO_PAIRINGS_FETCH_ERROR:
      return {...state, ...initialState};
    default:
      return state;
  }
};
