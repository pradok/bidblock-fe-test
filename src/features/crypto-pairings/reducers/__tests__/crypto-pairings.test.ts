import {cryptoPairings, initialState} from '../../reducers';
import {actionTypes as at} from '../../actions';
import {pairings} from '../../fixtures';

describe('cryptoPairings reducers', () => {
  it(`Action ${at.CRYPTO_PAIRINGS_FETCH} should return state`, () => {
    expect(
      cryptoPairings(initialState, {type: at.CRYPTO_PAIRINGS_FETCH})
    ).toMatchSnapshot()
  });
  it(`Action ${at.CRYPTO_PAIRINGS_FETCH_SUCCESS} should return state`, () => {
    expect(
      cryptoPairings(initialState, {type: at.CRYPTO_PAIRINGS_FETCH_SUCCESS, payload: pairings})
    ).toMatchSnapshot()
  });
  it(`Action ${at.CRYPTO_PAIRINGS_FETCH_ERROR} should return state`, () => {
    expect(
      cryptoPairings(initialState, {type: at.CRYPTO_PAIRINGS_FETCH_ERROR})
    ).toMatchSnapshot()
  });
});