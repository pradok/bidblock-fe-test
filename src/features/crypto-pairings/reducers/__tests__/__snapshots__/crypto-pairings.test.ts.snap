// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cryptoPairings reducers Action CRYPTO/PAIRINGS_FETCH should return state 1`] = `
Object {
  "collection": Array [],
  "isFetched": false,
  "isLoading": true,
  "tickerCollection": Array [],
}
`;

exports[`cryptoPairings reducers Action CRYPTO/PAIRINGS_FETCH_ERROR should return state 1`] = `
Object {
  "collection": Array [],
  "isFetched": false,
  "isLoading": false,
  "tickerCollection": Array [],
}
`;

exports[`cryptoPairings reducers Action CRYPTO/PAIRINGS_FETCH_SUCCESS should return state 1`] = `
Object {
  "collection": Array [
    Array [
      "tBTCUSD",
      6239.9,
      34.92908368,
      6240,
      35.7623858,
      -30.1,
      -0.0048,
      6239.9,
      18771.42426519,
      6339.9,
      6125,
    ],
    Array [
      "tLTCUSD",
      75.999,
      1099.6280602,
      76,
      486.87405918,
      -1.648,
      -0.0212,
      76,
      110391.68659301,
      78.66,
      75.049,
    ],
  ],
  "isFetched": true,
  "isLoading": false,
  "tickerCollection": Array [
    Object {
      "ask": 6240,
      "askSize": 35.7623858,
      "bid": 6239.9,
      "bidSize": 34.92908368,
      "dailyChange": -30.1,
      "dailyChangePerc": -0.0048,
      "high": 6339.9,
      "lastPrice": 6239.9,
      "low": 6125,
      "symbol": "tBTCUSD",
      "volume": 18771.42426519,
    },
    Object {
      "ask": 76,
      "askSize": 486.87405918,
      "bid": 75.999,
      "bidSize": 1099.6280602,
      "dailyChange": -1.648,
      "dailyChangePerc": -0.0212,
      "high": 78.66,
      "lastPrice": 76,
      "low": 75.049,
      "symbol": "tLTCUSD",
      "volume": 110391.68659301,
    },
  ],
}
`;
